File: native/swift/swift-export-standalone/testData/execution/exceptions/exceptions.kt
// KIND: STANDALONE
// MODULE: Main
// FILE: exceptions.kt

// FIXME: error: this annotation is not applicable to target 'getter'. Applicable targets: function, constructor
//val variableWithThrowingGetter: String
//    @Throws(Exception::class)
//    get() { error("variableWithThrowingGetter") }

@Throws(Exception::class)
fun throwingFunctionThatThrows(value: String): Any {
    error("fatal – $value")
}

class NonConstructible {
    @Throws(Exception::class)
    constructor(value: String) {
        error("fatal – $value")
    }
}

File: native/swift/swift-export-standalone-integration-tests/simple/testData/execution/exceptions/exceptions.swift
import Main
import Testing

@Test
func testThrowingFunction() throws {
    try #require(throws: Error.self) {
        try throwingFunctionThatThrows(value: "error")
    }
}

@Test
func testThrowingConstructor() throws {
    try #require(throws: Error.self) {
        try NonConstructible(value: "error")
    }
}